src/uucore/src/lib/features/fs.rs

```diff
@@ -70,6 +70,16 @@ pub enum CanonicalizeMode {
     Missing,
 }
 
+/// Controls when symbolic links are resolved
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ResolveMode {
+    /// Resolve symlinks as encountered when processing the path
+    Physical,
+
+    /// Resolve '..' elements before symlinks
+    Logical,
+}
```

Comment:
I might be totally wrong here and it's a remnant of the existing implementation, but I feel like the concerns are not well separated between `ResolveMode` and `CanonicalizeMode`. I would expect something like:
```rust
enum ResolveMode {
    None,
    Logical,
    Physical,
}

enum MissingHandling {
    Normal,
    Existing,
    Missing,
}
```

But maybe that doesn't make sense for a reason I'm not seeing...

Reply:
No, I think your suggestion is correct. The `CanonicalizeMode` is a remnant of the existing implementation and the options for handling symlinks should be grouped together.